wallet-new/server/Main.hs

```diff
@@ -1,211 +1,21 @@
-          bracketKernelPassiveWallet logMessage' keystore $ \walletLayer passiveWallet -> do
-            liftIO $ logMessage' Info "Wallet kernel initialized"
-            Kernel.Mode.runWalletMode pm
-                                      nr
-                                      walletLayer
-                                      (mainAction (walletLayer, passiveWallet) nr)
-  where
-    mainAction
-        :: (PassiveWalletLayer IO, PassiveWallet)
-        -> NodeResources ext
-        -> (Diffusion Kernel.Mode.WalletMode -> Kernel.Mode.WalletMode ())
-    mainAction w nr = runNodeWithInit w nr
-
-    runNodeWithInit
-        :: (PassiveWalletLayer IO, PassiveWallet)
-        -> NodeResources ext
-        -> (Diffusion Kernel.Mode.WalletMode -> Kernel.Mode.WalletMode ())
-    runNodeWithInit w nr = runNode pm nr (plugins w)
-
-    -- TODO: Don't know if we need any of the other plugins that are used
-    -- in the legacy wallet (see 'actionWithWallet').
-    plugins :: (PassiveWalletLayer IO, PassiveWallet)
-            -> Plugins.Plugin Kernel.Mode.WalletMode
-    plugins w = mconcat [ Plugins.walletBackend pm params w ]
-
-    -- Extract the logger name from node parameters
-    --
-    -- TODO: Not sure what the policy is for logger names of components.
-    -- For now we just use the one from the node itself.
-    logMessage' :: Severity -> Text -> IO ()
-    logMessage' sev txt =
-        usingLoggerName loggerName $ logMessage sev txt
-      where
-        loggerName :: LoggerName
-        loggerName = lpDefaultName . bpLoggingParams . npBaseParams $ nodeParams
-
--- | Runs an edge node plus its wallet backend API.
-startEdgeNode :: HasCompileInfo
-              => WalletStartupOptions
-              -> IO ()
-startEdgeNode wso =
-  withConfigurations blPath conf $ \ntpConfig pm -> do
-      (sscParams, nodeParams) <- getParameters ntpConfig
-      case wsoWalletBackendParams wso of
-        WalletLegacy legacyParams ->
-          actionWithWallet pm sscParams nodeParams ntpConfig legacyParams
-        WalletNew newParams ->
-          actionWithNewWallet pm sscParams nodeParams newParams
-  where
-    getParameters :: HasConfigurations => NtpConfiguration -> IO (SscParams, NodeParams)
-    getParameters ntpConfig = do
-
-      currentParams <- CLI.getNodeParams defaultLoggerName (wsoNodeArgs wso) nodeArgs
-      let vssSK = fromJust $ npUserSecret currentParams ^. usVss
-      let gtParams = CLI.gtSscParams (wsoNodeArgs wso) vssSK (npBehaviorConfig currentParams)
-
-      CLI.printInfoOnStart (wsoNodeArgs wso) ntpConfig
-      logInfo "Wallet is enabled!"
-
-      return (gtParams, currentParams)
-
-    conf :: ConfigurationOptions
-    conf = CLI.configurationOptions $ CLI.commonArgs (wsoNodeArgs wso)
-
-    blPath :: Maybe AssetLockPath
-    blPath = AssetLockPath <$> CLI.cnaAssetLockPath (wsoNodeArgs wso)
-
-    nodeArgs :: CLI.NodeArgs
-    nodeArgs = CLI.NodeArgs { CLI.behaviorConfigPath = Nothing }
-
```

Comment:
This has been moved away and slightly adjusted in `Cardano/Wallet/Launcher`. Changes are mostly in the function APIs that have been designed to work with `NodeArgs` and `CommonNodeArgs` instead of already instantiated stuff. This way, we can keep the calling code clean, and take care of allocating resources and opening connections all in one place in the `start***Node` functions.